docs/excel/custom-functions-naming.md

```diff
@@ -0,0 +1,38 @@
+---
+ms.date: 02/08/2019
+description: Learn requirements for Excel custom functions' names and avoid common naming pitfalls.
+title: Naming guidelines for custom functions in Excel (preview)
+localization_priority: Normal
+---
+# Naming guidelines
+
+Custom functions contain two unique properties in their JSON metadata file: their function name and their function ID. Function IDs are used as the main way to differentiate functions from each other in Excel programmatically. Function names are what appear to users of your add-in as the function's name in Excel. Function names can differ from the ID, e.g. for localization purposes, but should generally stay the same as the ID if there is not a compelling reason for them to differ.
+
+Function names and function IDs share some common requirements:
+
+- Can use alphanumeric characters (including Unicode), the numbers zero through nine, underscores, and periods.
+
+- Must start with a letter and have a minimum limit of three characters.
+
+Most function names for built-in Excel functions appear in uppercase letters (e.g. `SUM`). While not required, it is a best practice to specify your custom function names and function IDs using uppercase letters.
+
+Function names should not be named the same as:
```

Comment:
won't this cause some kind of error or just unexpected behavior? We should document just so this doesn't appear as an optional kind of thing.

Reply:
So as far as I know it actually doesn't cause an error - ID automatically converts to be uppercase and name is something that doesn't care about case. We want people to use uppercase because it matches what we do in Excel, but they don't technically have to. I could take out "While not required"?

Reply:
got it, this looks good then.

Reply:
I agree with Michelle, it should be a best practices. I would simply say "Most function names for built-in Excel functions appear in uppercase letters (e.g. "SUM"). Therefore, it is a best practice  to use uppercase letters for your custom function names and function IDs."

Reply:
I made this in a "suggestion" elsewhere